### ISSUE TITLE
Latest version on capacitor not returning any data

### ISSUE THREAD

**Issue by Saqib92** [dariosalvi78/cordova-plugin-health#353]
Latest version on capacitor not returning any data always return empty Array.

**Comment by Saqib92** [comment]
any update?

**Comment by Saqib92** [comment]
any update on this?

**Comment by Asko-Dev** [comment]
I also don't know if that's samsung health problem, it seems to not be writing anything to health connect but I get empty arrays with query() and with queryAggregated I just get ~1564 kcal every time for basal and total calories, no other amount, despite having completely different numbers in samsung health... but again .... seems more like a samsung health problem

**Comment by dariosalvi78** [comment]
@Asko-Dev  are you sure that Samsung Health is linked with HealthConnect?

**Comment by Asko-Dev** [comment]
yes I am ... it's even in their "list of apps u can connect to us" .. it now started properly sending data I see the logs. Finally logging activity and body composition. I had to reinstall samsung health, then started walking around to gain steps... the write logs are visible in health connect now. 
![logs](https://github.com/user-attachments/assets/7c5baaeb-c704-434c-99d9-9b027d82358f)

but unfortunately ... query() just returns empty arrays and aggregate (has correct dates) just returns what seems to be a default value of some sort 1564 kcal, I saw it somewhere online some guy was reporting the same value coming from `_platform` I will try to find the link, unfortunately just glossed over it at the time.

There is a known issue that Samsung health does not allow read/ write of active calories so I resorted it to basal and total in hopes to deduct basal from total. But those two are returning empty arrays with .query() and that default value 1564 with aggregate .... none of those values correspond with what samsung health is showing in the app.

But it rly could be samsung health issue

**Comment by Asko-Dev** [comment]
steps are querying ... I do get steps .... and correct amount even, so maybe samsung health is not writing calories?

**Comment by Asko-Dev** [comment]
ok definitely not your issue, finally found a piece of information, they do not sync activity calories, only exercise, I just tried to turn on walking exercise and walked and that worked perfectly, but no steps calories :/

**Comment by Saqib92** [comment]
I am not using Samsung health but still there is no data from any query.

**Comment by Saqib92** [comment]
@dariosalvi78 any update? new version `"cordova-plugin-health": "^3.2.4",` returns empty array for any query. steps, calories etc.
